enlace/ajax/add_date.php

```diff
@@ -33,6 +33,36 @@
     mysqli_query($cnnEnlace, $delete_date);
     include "../include/dbconnclose.php";
 }
+else if ($_POST['action']=='generate'){
+    include "../include/dbconnopen.php";
+    error_log("Generating, yo!");
+    $session_id_sqlsafe=mysqli_real_escape_string($cnnEnlace, $_POST['session_id']);
+    $session_query = "SELECT * FROM Session_Names WHERE Session_ID='$session_id_sqlsafe'";
+    $session_info = mysqli_fetch_row(mysqli_query($cnnEnlace, $session_query));
+    if($session_info !== FALSE && $session_info[3] != '0000-00-00' && $session_info[4] != '0000-00-00' &&
+       $session_info[3] < $session_info[4]) {
+        $program_query = "SELECT * FROM Programs WHERE Program_ID='".$session_info[2]."'";
+        $program_info = mysqli_fetch_array(mysqli_query($cnnEnlace, $program_query));
+        date_default_timezone_set('America/Chicago');
+        $session_start = new DateTime($session_info[3]);
+        $session_end = new DateTime($session_info[4]);
+        $date_to_add = $session_start;
+        $interval = new DateInterval('P1D');
+        while($date_to_add < $session_end) {
+            // These match the database exactly, matching a format output from DateTime->format is coincedental and didn't
+            // want to count on it.
+            $days = array("Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday");
+
+            if($program_info[$days[$date_to_add->format('w')]] == 1) {
+                $new_date="INSERT INTO Program_Dates (Program_ID, Date_Listed) VALUES
+                    ('".$session_id_sqlsafe."', '".$date_to_add->format('Y-m-d')."')";
+                mysqli_query($cnnEnlace, $new_date);
+            }

```

Comment:
If the program has no days of the week chosen this will fail silently.  It needs a catch for the case where the user hasn't entered any days of the week, otherwise the "generate attendance" appears to work, but creates no dates -- very confusing for the user!  I know this from personal experience because I tried to create a new program to test with, and accidentally neglected to give it days of the week. :woman_facepalming:  The "create new program" dialogue on the "Programs" page doesn't offer a place for days of the week, so users have to go in and add them separately.  It might even be good to do a check for this early and show a warning/disable the "generate attendance" link if there are no days of the week chosen.